pkg/controller/util/node/controller_utils.go

```diff
@@ -133,6 +138,11 @@ func MarkPodsNotReady(kubeClient clientset.Interface, pods []*v1.Pod, nodeName s
 				klog.V(2).Infof("Updating ready status of pod %v to false", pod.Name)
 				_, err := kubeClient.CoreV1().Pods(pod.Namespace).UpdateStatus(pod)
 				if err != nil {
+					if apierrors.IsNotFound(err) {
+						// Not founding pods is something that can happen in a cluster.
+						// This handling is crucial for correctness.
```

Comment:
Same here - it's not obvious why it is crucial for correctness.

// NotFound error means that pod was already deleted. There is nothing left to do with this pod.

Reply:
Done.